icons.go

```diff
@@ -4,10 +4,26 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"regexp"
 	"strings"
 )
 
-type iconMap map[string]string
+// Icons that can be matched through a simple string lookup
+type basicIcon struct {
+	icon string
+	pos  int
+}
+
+// Icons that need to be matched (globbed) to classify
+type globIcon struct {
+	pattern *regexp.Regexp
+	basicIcon
+}
+
+type iconMap struct {
+	basicIcons map[string]basicIcon
+	globIcons  []globIcon
+}
```

Comment:
Probably worth noting there is a memory overhead here. I needed to keep track of the position of entries in the original LF_ICONS variable to avoid redundant matches. `*.txt=foo:*t*=bar` no point trying to match the second pattern when the first is always going to be used.

Do you think there's any way to optimize this?